be/src/olap/rowset/segment_v2/binary_plain_page.h

```diff
@@ -81,24 +81,19 @@ class BinaryPlainPageBuilder : public PageBuilder {
     Slice finish() override {
         _finished = true;
 
-        size_t offsets_pos = _buffer.size();
-
-        // Set up the header
-        encode_fixed32_le(&_buffer[0], _offsets.size());
-        encode_fixed32_le(&_buffer[4], offsets_pos);
-
-        _buffer.append(&_offsets[0], _offsets.size() * sizeof(uint32_t));
+        // Set up trailer
+        for (int i = 0;i < _offsets.size();i ++) {
+            put_fixed32_le(&_buffer, _offsets[i]);
+        }
+        put_fixed32_le(&_buffer, _offsets.size());
 
-        return Slice(_buffer.data(), _buffer.size());
+        return Slice(_buffer);
     }
 
     void reset() override {
         _offsets.clear();
         _buffer.clear();
-        _buffer.resize(MAX_HEADER_SIZE);
         _buffer.reserve(_options.data_page_size);
```

Comment:
_options.data_page_size could be 0 when there is no limit for page size, in that case we'd better reserve more than 0 bytes (1024 looks good to me)